python/tests/test_metrics.py

```diff
@@ -0,0 +1,97 @@
+#
+# Copyright (c) 2023, NVIDIA CORPORATION.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+import numpy as np
+import pandas as pd
+import pytest
+from pyspark.ml.evaluation import MulticlassClassificationEvaluator
+
+from python.tests.sparksession import CleanSparkSession
+from spark_rapids_ml.metrics.MulticlassMetrics import MulticlassMetrics
+
+
+def get_multi_class_metrics(pdf: pd.DataFrame, num_classes: int) -> MulticlassMetrics:
+    confusion = pdf.groupby(["label", "prediction"]).size().reset_index(name="total")
+
+    tp_by_class = {}
+    fp_by_class = {}
+    label_count_by_class = {}
+    label_count = 0
+
+    for i in range(num_classes):
+        tp_by_class[float(i)] = 0.0
+        label_count_by_class[float(i)] = 0.0
+        fp_by_class[float(i)] = 0.0
+
+    for index, row in confusion.iterrows():
+        label_count += row.total
+        label_count_by_class[row.label] += row.total
+
+        if row.label == row.prediction:
+            tp_by_class[row.label] += row.total
+        else:
+            fp_by_class[row.prediction] += row.total
+
+    return MulticlassMetrics(
+        tp=tp_by_class,
+        fp=fp_by_class,
+        label=label_count_by_class,
+        label_count=label_count,
+    )
+
+
+@pytest.mark.parametrize("num_classes", [4])
+@pytest.mark.parametrize(
+    "metric_name",
+    [
+        "f1",
```

Comment:
Looks like a lot of cases, is the test taking a long time?

Reply:
just a few seconds. there is no training/inference, so they're running quite fast.